users/views.py

```diff
@@ -1,3 +1,119 @@
+                else:
+                    error = 'The account has been disabled.'
+                    return render(request, 'users/login.html',
+                                  {'error': error})
+            else:
+                error = 'Invalid Username/Password'
+                return render(request, 'users/login.html', {'error': error})
+        else:
+            return render(request, 'users/login.html', {})
+
+
+def logout(request):
+    auth_logout(request)
+    return redirect(reverse('users:login'))
+
+
+def register(request):
+    if request.user.is_authenticated:
+        return render(request, 'users/test.html', {})
+    else:
+        if request.method == 'POST':
+            # Username and email fields will have the same data
+            username = request.POST.get('email', '')
+            password = request.POST.get('password', '')
+            sap_id = request.POST.get('sap_id', '')
+            mobile = request.POST.get('mobile')
+            errors = {}
+            # Check if no other user has the same email id
+            if CustomUser.objects.filter(username=username).exists():
+                errors['email_error'] = 'The email is already in use by another account.'
+            # Check for uniqueness of SAP ID
+            if CustomUser.objects.filter(sap_id=sap_id).exists():
+                errors['sap_error'] = 'The SAP ID is already in use by another account.'
+            # Check for uniqueness of Mobile No.
+            if CustomUser.objects.filter(mobile=mobile).exists():
+                errors['mobile_error'] = 'The mobile number is already in use by another account.'
+            if len(errors) > 0:
+                return render(request, 'users/register.html', errors)
+            else:
+                email = request.POST.get('email', '')
+                first_name = request.POST.get('first_name', '')
+                last_name = request.POST.get('last_name', '')
+                user = CustomUser.objects.create(username=username, email=email, sap_id=sap_id, mobile=mobile,
+                                                 first_name=first_name, last_name=last_name)
+                user.is_superuser = False
+                user.is_staff = False
+                user.set_password(password)
+                user.save()
+                auth_login(request, user)
+                return redirect('users:update_profile', pk=user.id)
+        else:
+            return render(request, 'users/register.html', {})
+
+
+def view_profile(request, pk):
+    if not request.user.is_authenticated:
+        return render(request, 'users/login.html', {})
+    else:
+        user = get_object_or_404(CustomUser, pk=pk)
+        return render(request, 'users/profile.html', {'user': user})
+
+
+def update_profile(request, pk):
+    if not request.user.is_authenticated:
+        return render(request, 'users/login.html', {})
+    elif request.user.pk != pk:
```

Comment:
An user is already logged in and without logging the user out, you are rendering the login page again. Doesn't seem good.

Reply:
I was talking about line 86 and 87. Even though the user is already is logged in, the login page is being rendered again. If you redirect to login here, it should solve the problem.